frontend/app/src/components/Sidebar/Sidebar.tsx

```diff
@@ -67,68 +67,100 @@ export interface SidebarProps {
   expandSidebarNav: boolean
 }
 
-interface State {
-  collapsedSidebar: boolean
-  sidebarWidth: string
-  lastInnerWidth: number
+const MIN_WIDTH = "336"
 
-  // When hovering sidebar header
-  showSidebarCollapse: boolean
+function calculateMaxBreakpoint(value: string): number {
+  // We subtract a margin of 0.02 to use as a max-width
+  return parseInt(value, 10) - 0.02
 }
 
-class Sidebar extends PureComponent<SidebarProps, State> {
-  private mediumBreakpointPx: number
+const Sidebar: React.FC<SidebarProps> = ({
+  theme,
+  appLogo,
+  endpoints,
+  appPages,
+  chevronDownshift,
+  children,
+  initialSidebarState,
+  hasElements,
+  onPageChange,
+  currentPageScriptHash,
+  hideSidebarNav,
+  expandSidebarNav,
+  navSections,
+}) => {
+  const mediumBreakpointPx = calculateMaxBreakpoint(theme.breakpoints.md)
+
+  const sidebarRef = useRef<HTMLDivElement>(null)
+
+  const cachedSidebarWidth = localStorageAvailable()
+    ? window.localStorage.getItem("sidebarWidth")
+    : undefined
```

Comment:
I think it would be better to move this to a function outside of the component and then call that function once in [line 104](https://github.com/streamlit/streamlit/pull/10230/files#diff-2266d533f4d77a235333ebe3f96a6fb8d49928f2444da02c1f8a12552744492dR104). This should guarantee the previous behavior where it is only called once during first render.
If it is extracted into a standalone function, also a unit test could be added for the function itself more easily.